akka-stream/src/main/scala/akka/stream/impl/SynchronousFilePublisher.scala

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import java.io.{ File, RandomAccessFile }
+import java.nio.channels.FileChannel
+
+import akka.actor.{ DeadLetterSuppression, ActorLogging, Props }
+import akka.io.DirectByteBufferPool
+import akka.stream.actor.ActorPublisherMessage
+import akka.util.ByteString
+
+import scala.concurrent.Promise
+
+/** INTERNAL API */
+private[akka] object SynchronousFilePublisher {
+  def props(f: File, completionPromise: Promise[Long], chunkSize: Int, initialBuffer: Int, maxBuffer: Int) = {
+    require(chunkSize > 0, s"chunkSize must be > 0 (was $chunkSize)")
+    require(initialBuffer > 0, s"initialBuffer must be > 0 (was $initialBuffer)")
+    require(maxBuffer >= initialBuffer, s"maxBuffer must be >= initialBuffer (was $maxBuffer)")
+
+    Props(classOf[SynchronousFilePublisher], f, completionPromise, chunkSize, initialBuffer, maxBuffer)
+  }
+
+  private final case object Continue extends DeadLetterSuppression
+}
+
+/** INTERNAL API */
+private[akka] class SynchronousFilePublisher(f: File, bytesReadPromise: Promise[Long], chunkSize: Int, initialBuffer: Int, maxBuffer: Int)
+  extends akka.stream.actor.ActorPublisher[ByteString]
+  with ActorLogging {
+
+  import SynchronousFilePublisher._
+
+  val buffs = new DirectByteBufferPool(chunkSize, maxBuffer)
```

Comment:
I'm torn. I understand your arguments @jrudolph yet am unable to prove it with a benchmark, currently I'm at (for reading a ~500M file):

```
[info] Benchmark                         (bufSize)  Mode  Cnt    Score     Error  Units
[info] FileSourcesBenchmark.fileChannel       2048  avgt   80  591.803 ±  68.584  ms/op
[info] FileSourcesBenchmark.inputStream       2048  avgt   80  714.225 ± 106.160  ms/op
```
See this commit for code and bench: https://github.com/ktoso/akka/commit/fe822de517845819219b99a1356da7caaae898fe

It's also a question on API - I guess we do want to expose an "file" API, independently how it's implemented. The Input / Output Streams would be additional `Sink.outputStream` I guess? (some of those exist already https://github.com/typesafehub/akka-contrib-extra/tree/master/src/main/scala/akka/contrib/stream )